x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -420,19 +353,17 @@ export class Authenticator {
   async logout(request: KibanaRequest) {
     assertRequest(request);
 
-    const sessionStorage = this.options.sessionStorageFactory.asScoped(request);
-    const sessionValue = await this.getSessionValue(sessionStorage);
+    const sessionValue = await this.session.get(request);
     if (sessionValue) {
-      sessionStorage.clear();
-
+      await this.session.clear(request);
       return this.providers.get(sessionValue.provider.name)!.logout(request, sessionValue.state);
     }
 
-    const providerName = this.getProviderName(request.query);
-    if (providerName) {
+    const queryStringProviderName = (request.query as Record<string, string>)?.provider;
+    if (typeof queryStringProviderName === 'string') {
```

Comment:
__note:__ hmm, can't recall why I changed this, I'll revert back to `if (queryStringProviderName) {`.